retro_video.py

```diff
@@ -0,0 +1,259 @@
+        if not name:
+            name = self.selected_customer["name"]
+        if not phone:
+            phone = self.selected_customer["phone"]
+        if not postal_code:
+            postal_code = self.selected_customer["postal_code"]
+        if not registered_at:
+            registered_at = self.selected_customer["registered_at"]
+        if not videos_checked_out_count:
+            videos_checked_out_count = self.selected_customer["videos_checked_out_count"]
+
+        query_params = {
+            "name": name,
+            "phone": phone,
+            "postal_code": postal_code,
+            "registered_at": registered_at,
+            "videos_checked_out_count": videos_checked_out_count
+            }
+        response = requests.put(
+            self.url+f"/customers/{self.selected_customer['id']}",
+            json=query_params
+            )
+        print("response:", response)
+        self.selected_customer = response.json()
+        return response.json()
+
+###### "8": "delete a customer" DELETE CUSTOMER #####
+    def delete_customer(self):
+        print("\n❌ 📼 ❌ THE CUSTOMER YOU SELECT WILL BE DELETED ❌ 📼 ❌\n")
+        self.select_customer()
+        response = requests.delete(self.url+f"/customers/{self.selected_customer['id']}")
+        self.selected_customer = None
+        return response.json()
+
+###### "9": "get information about one customer" READ CUSTOMER #####
+    def get_customer(self, name=None, id=None):
+        for customer in self.list_customers():
+            if customer:
+                if customer["name"]==name:
+                    id = customer["id"]
+                    self.selected_customer = customer
+                elif id == customer["id"]:
+                    self.selected_customer = customer
+
+        if self.selected_customer == None:
+            return "🤷 Could not find customer by that name or id"
+
+        response = requests.get(self.url+f"/customers/{id}")
+        return response.json()
+
+###### "10": "get information about all customers" READ CUSTOMER #####
+    def list_customers(self):
+        response = requests.get(self.url+"/customers")
+        return response.json()
+
+#######################################
+########### RENTAL OPTIONS ############
+#######################################
+
+###### "11": "check out a video to a customer" CHECK OUT #####
+    def check_out(self, customer_id="Default Customer_ID",video_id="Default Video_ID"):
+        self.select_customer()
+        self.select_video()
+
+        customer_id = self.selected_customer["id"]
+        video_id = self.selected_video["id"]
+
+        video_title = self.selected_video["title"]
+        customer_name = self.selected_customer["name"]
+
+        query_params = {
+            "customer_id": customer_id,
+            "video_id": video_id
+            }
+        response = requests.post(self.url+f"/rentals/check-out", json=query_params)
+        rental_due_date = response.json()['due_date']
```

Comment:
If the checkout fails, this will throw an error.  Checking the response.status_code is one way to validate that the request was successful before moving on.